lib/dbservice_web/controllers/enrollment_record_controller.ex

```diff
@@ -74,15 +75,21 @@ defmodule DbserviceWeb.EnrollmentRecordController do
   end
 
   def create(conn, params) do
-    with {:ok, %EnrollmentRecord{} = enrollment_record} <-
-           Schools.create_enrollment_record(params) do
-      conn
-      |> put_status(:created)
-      |> put_resp_header(
-        "location",
-        Routes.enrollment_record_path(conn, :show, enrollment_record)
-      )
-      |> render("show.json", enrollment_record: enrollment_record)
+    existing_record =
+      index(conn, %{
+        "student_id" => params["student_id"],
+        "school_id" => params["school_id"],
+        "group_id" => params["group_id"],
+        "group_type" => params["group_type"],
+        "academic_year" => params["academic_year"]
+      })
```

Comment:
Or maybe we can directly pass the params?
Something like:
```suggestion
      index(conn, params)
```

Reply:
No, no, if we pass the whole params, all the parameters will be checked. We want only these specific ones to be checked.

Reply:
Index function will not return `existing_record` here. This seems an invalid approach (maybe you can also test it out to have more clarity). I would recommend to create a function in `school.exs` to handle this filter.